state_synchronizer/src/coordinator.rs

```diff
@@ -328,27 +330,25 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
 
         let previous_version = self.known_version;
 
-        match LedgerInfoWithSignatures::<Ed25519Signature>::from_proto(
+        let target = LedgerInfoWithSignatures::<Ed25519Signature>::from_proto(
             response.take_ledger_info_with_sigs(),
-        ) {
-            Ok(target) => {
-                if let Err(err) = self.store_transactions(target, txn_list_with_proof).await {
-                    error!("[state sync] failed to apply chunk {:?}", err);
-                }
-                counters::STATE_SYNC_TXN_REPLAYED.inc_by(chunk_size as i64);
-                match self.executor_proxy.get_latest_version().await {
-                    Ok(version) => {
-                        self.commit(version).await;
-                    }
-                    Err(err) => {
-                        error!("[state sync] storage version read failed {:?}", err);
-                    }
-                }
+        )?;
+        self.executor_proxy.validate_ledger_info(&target)?;
+
+        if let Err(err) = self.store_transactions(target, txn_list_with_proof).await {
```

Comment:
nit: no need to transform it. Just use same syntax `self.store_transactions(..)?`